pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2027,6 +2016,33 @@ func buildTracingConfig(config *meshconfig.ProxyConfig, proxyType model.NodeType
 	return tracingCfg
 }
 
+func getPilotRandomSamplingEnv() float64 {
+	f := features.TraceSampling
+	if f < 0.0 || f > 100.0 {
+		log.Warnf("PILOT_TRACE_SAMPLING out of range: %v", f)
```

Comment:
Why it this not validated at boot time and then default to something sane when the ENV value is out of range?

@howardjohn any reason we shouldn’t move this validation to bootstrap?

Reply:
Not a fan of of package variables. There is a precedent in the listener code and before my first PR tracing was built and stored in a [package variable](https://github.com/istio/istio/pull/22880/files#diff-86fb73d7dd150e47328ead76ceca0992L333). Since it is derived from an environmental variable it makes sense in this case, especially with validation and a log.